sdk/monitor/monitor-ingestion/review/monitor-ingestion.api.md

```diff
@@ -0,0 +1,54 @@
+## API Report File for "@azure/monitor-ingestion"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { CommonClientOptions } from '@azure/core-client';
+import { TokenCredential } from '@azure/core-auth';
+
+// @public (undocumented)
+export class FailedLogsIngestionError extends Error {
+    constructor(message: string, errors: Array<UploadLogsError>);
+    // (undocumented)
+    errors: Array<UploadLogsError>;
+}
+
+// @public
+export class LogsIngestionClient {
+    constructor(endpoint: string, tokenCredential: TokenCredential, options?: LogsIngestionClientOptions);
+    // @internal (undocumented)
+    splitDataToChunks(logs: Record<string, any>[]): any[];
+    upload(ruleId: string, streamName: string, logs: Record<string, any>[], options?: UploadOptions): Promise<UploadResult>;
```

Comment:
Can we get a slightly more specific name for this? Also would be good to use `unknown` I think:
`
``suggestion
    uploadLogs(ruleId: string, streamName: string, logs: Record<string, unknown>[], options?: UploadOptions): Promise<UploadResult>;
```

Reply:
The client name is "LogsIngestionClient". If i use "logs" in the method name, won't that be redundant?

Reply:
If I switch the `any` to `unknown` I get this error when I try to pass in some structured data type - `Argument of type 'LogData[]' is not assignable to parameter of type 'Record<string, unknown>[]'`

Reply:
@xirzec

Reply:
Line 80, 98 and 121 of logsIngestion.spec.ts

Reply:
It looks like typescript is having issues assigning a less generic interface to a more generic one. This gest solved in the test if the LogData interface is defined as a type or if the interface extends Record<string, unknown>. Inlining the object seems to work fine with `Record<string, unknown>`

Reply:
Yes, thanks @joheredi. The TS error gets resolved when I switch to `type` instead of `interface` for `LogData`. Though would the customers be aware of this?

Reply:
Scott Addie has opened a poll on twitter for making a decision for `upload` v/s `uploadLogs` https://twitter.com/Scott_Addie/status/1547956833003810820